test/ui/test_plot2d_dialog.py

```diff
@@ -0,0 +1,97 @@
+import os
+import pytest
+from RefRed.main import MainGui
+from qtpy import QtCore, QtWidgets
+
+
+wait = 100
+
+
+def test_plot2d_dialog(qtbot, data_server):
+    """Test the plot2d dialog."""
+    # Create the main window
+    window = MainGui()
+    qtbot.addWidget(window)
+
+    # window.show()  # Only for human inspection. This line should be commented once the test passes.
+
+    # Load data
+    xmlfilename = data_server.path_to("REF_L_188298_tiny_template.xml")
+    window.path_config = os.path.dirname(xmlfilename)
+
+    def qfiledialog_handler():
+        def handler():
+            dialog = window.findChild(QtWidgets.QFileDialog)
+            line_edit = dialog.findChild(QtWidgets.QLineEdit)
+            qtbot.keyClicks(line_edit, os.path.basename(xmlfilename))
+            qtbot.wait(wait)
+            qtbot.keyClick(line_edit, QtCore.Qt.Key_Enter)
+
+        QtCore.QTimer.singleShot(5 * wait, handler)  # wait for `lapse` time, then execute `handler`
+
+    qfiledialog_handler()
+    window.load_configuration()
+    qtbot.wait(wait)
+
+    # Toggle the plot up
+    window.reduction_table_visibility_changed_test(1, 0)
+    qtbot.wait(wait)
+
+    # Trigger the plot2d dialog
+    # NOTE: somehow RefRed is not utilizing the built-in double
```

Comment:
Wouldn't it make more sense to replace whatever quirky mechanism is currently implemented to the "built-in double click"?

Reply:
so that about this?